src/Symfony/Component/RateLimiter/RateLimiterFactory.php

```diff
@@ -41,15 +41,19 @@ public function __construct(array $config, StorageInterface $storage, LockFactor
         $this->config = $options->resolve($config);
     }
 
-    public function create(string $key = null): LimiterInterface
+    /**
+     * Parsing a custom limit to this function will overwrite the limit in your configuration for this request.
+     * When using a NoLimiter the limit will be ignored.
+     */
+    public function create(string $key = null, int $limit = null): LimiterInterface
```

Comment:
What if you also want at some point to configure the rate dynamically? A solution could be to provide the whole config instead (and resolve it), WDYT? 

```suggestion
    public function create(string $key = null, array $config = null): LimiterInterface
```

Reply:
this would then force to validate the provided config each time this method is called.

To me, if you want to do things dynamically, it would be better to introduce a RateLimiterFactoryInterface allowing to do a custom implementation of the factory than adding a configuration override argument.
Thus, if you are creating a custom caller for the factory, you could already instantiate the limiter yourselves. And if you don't call it yourselves (relying on a usage in Symfony), you would not be able to pass a custom config.

Reply:
Indeed, I that if we add an argument to the factory method to customize it, there is no point to not adding others (even if the limit is the only common parameter).

That's why, in my opinion too, it is way better to indeed introduce a `RateLimiterFactoryInterface` so there is no limit to the customization.

Reply:
Alright :) I will create the interface instead later this week. Thanks guys!